pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -118,28 +129,31 @@ func VerifyDeviceNetworkStatus(status types.DeviceNetworkStatus,
 		if err != nil {
 			errStr := "Onboarding certificate cannot be found"
 			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			return false, errors.New(errStr), intfErrMap
 		}
 		clientCert := &onboardingCert
 		tlsConfig, err = zedcloud.GetTlsConfig(zedcloudCtx.DeviceNetworkStatus,
 			serverName, clientCert, &zedcloudCtx)
 		if err != nil {
-			errStr := fmt.Sprintf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
-
-			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			err = fmt.Errorf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
+			log.Infof("VerifyDeviceNetworkStatus: %s\n", err)
+			return false, err, intfErrMap
 		}
 	}
 	zedcloudCtx.TlsConfig = tlsConfig
 	for ix := range status.Ports {
 		err = CheckAndGetNetworkProxy(&status, &status.Ports[ix])
 		if err != nil {
-			errStr := fmt.Sprintf("GetNetworkProxy failed %s", err)
-			log.Errorf("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			ifName := status.Ports[ix].IfName
+			err = fmt.Errorf("GetNetworkProxy: ifName: %s, failed %s",
+				ifName, err)
+			log.Errorf("VerifyDeviceNetworkStatus: %s\n", err)
+			intfErrMap[ifName] = types.NewErrorInfo(err, "")
+			return false, err, intfErrMap
```

Comment:
Not part of your code change but it is odd that the old code failed here, as opposed to continuing iterating over the ports.

If we have failures from CheckAndGetNetworkProxy from all of the ports, then we should give up. But otherwise have VerifyAllIntf try the other ports. Maybe I'm missing something

Reply:
I am not particularly familiar with this code path.. If you think it is ok to cycle through all the ports here, I can make the change..

Reply:
Hmm.. Looking at the code - it sure sounds like we should circle through all the ports..